examples/usematches-loader-data/app/useMatchesData.ts

```diff
@@ -0,0 +1,17 @@
+import { useMemo } from "react";
+import { useMatches } from "remix";
+
+/**
+ * This base hook is used in other hooks to quickly search for specific data
+ * across all loader data using useMatches.
+ * @param {string} id The route id
+ * @returns {JSON} The value of the key based on your currently loaded routes.
+ */
+export default function useMatchesData(id: string): Record<string, unknown> {
+  const matchingRoutes = useMatches();
+  const route = useMemo(() => matchingRoutes.find((route) => route.id === id), [
+    matchingRoutes,
+    id
+  ]);
+  return route?.data || {};
+}
```

Comment:
```suggestion
export default function useMatchesData(id: string): Record<string, unknown> | undefined {
  const matchingRoutes = useMatches();
  const route = useMemo(() => matchingRoutes.find((route) => route.id === id), [
    matchingRoutes,
    id
  ]);
  return route?.data;
}
```

Reply:
> Thanks for the back-and-forth. I think we're getting close to a terrific solution 👍

Thank you for taking the time! `useMatches` is awesome but it's low-level, so it's great that we reiterate on this particular usage example! I really appreciate the feedback! :) 

Snap! I had issues because `unknown` was complaining but I already fixed it by using `Record<string, unknown>`, so there was really no need to avoid the undefined anymore! Thanks for the change request!